streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -915,6 +917,13 @@ public void punctuate(final ProcessorNode<?, ?, ?, ?> node,
 
         try {
             maybeMeasureLatency(() -> punctuator.punctuate(timestamp), time, punctuateLatencySensor);
+        } catch (final TimeoutException timeoutException) {
```

Comment:
Totally unrelated (might actually to good to put into different PR, but want to avoid that I forget it).

Seems we forgot to handle this case entirely... we might call `context.forward()` inside of  `punctuate()` and thus need to handle the same error cases as for `node.process()`...

Reply:
> Seems we forgot to handle this case entirely... we might call context.forward() inside of punctuate() and thus need to handle the same error cases as for node.process()...

We could end up in a sink node, and thus in `RecordCollectorImpl` / `producer.send()` which could throw a `TimeoutException`.